### PR TITLE
Fix 1383 Llama model on transformers=4.41[WIP]

### PR DESCRIPTION
**Pull request by songhappy**
## Description
add llama_attention_forward_4_41, llama_model_forward_4_41

### CODE REVIEW

**Comment by songhappy** [comment]
Tested on Max1100 and documented LLama2-7B model metrics on [issue-1383](https://github.com/analytics-zoo/nano/issues/1383), performance metrics of transformers4.41 are similar to 4.38

File: python/llm/src/ipex_llm/transformers/models/llama.py [hunk 1]
```diff
@@ -1483,81 +1519,604 @@ def llama_attention_forward_4_38_original(
...
+            rope_theta = self.rotary_emb.base
+            query_states, key_states = apply_rotary_pos_emb_no_cache_xpu(query_states,
+                                                                         key_states,
+                                                                         position_ids,
+                                                                         "llama",
+                                                                         rope_theta=rope_theta)
+        else:
             if cache_position is not None:
                 # for transformers 4.38.0
-                causal_mask = attention_mask[:, :, cache_position, : kv_seq_len]
```
Comment (sgwhat) [inline python/llm/src/ipex_llm/transformers/models/llama.py:1504]: The reason to remove `causal_mask` here?

File: python/llm/src/ipex_llm/transformers/models/llama.py [hunk 2]
```diff
@@ -1839,6 +2399,134 @@ def llama_attention_fast_forward(
...
+
+    hidden_states = self.norm(hidden_states)
+
+    # add hidden states from the last decoder layer
+    if output_hidden_states:
+        all_hidden_states += (hidden_states,)
+
+    next_cache = next_decoder_cache if use_cache else None
+    if return_legacy_cache:
+        next_cache = next_cache.to_legacy_cache()
```
Comment (sgwhat) [inline python/llm/src/ipex_llm/transformers/models/llama.py:2517]: Need to double check if `next_decoder_cache` is `DynamicFP8Cache`.

**Review by sgwhat — APPROVED** [review]
Others LGTM

File: python/llm/src/ipex_llm/transformers/models/llama.py (hunk 1 above)
Comment (songhappy) [reply python/llm/src/ipex_llm/transformers/models/llama.py:1504]: the code is compared to wrong place, did not touch for 4_38

**Pull request merged**